### PR TITLE
Improve error message on unbound args

### LINKED ISSUE #87: Test unbound arguments false positive in presence of vararg and typeassert
```julia
module MMM
f(xs::Vector{T}...) where {T<:Integer} = [one(T)*x for x in xs] # raises an error
g(x::Vector{T}) where {T<:Integer} = one(T)*x
h(xs...)= [one(eltype(x))*x for x in xs]
i(x) = one(eltype(x))*x
end

using Aqua

Aqua.test_unbound_args(MMM)
```

raises an error:

```
Test Failed at /home/stefan/.julia/packages/Aqua/HWLbM/src/unbound_args.jl:10
  Expression: detect_unbound_args_recursively(m) == []
   Evaluated: Any[f(xs::Vector{T}...) where T<:Integer @ Main.MMM scratch.jl:4] == Any[]
ERROR: There was an error during testing
```

```
julia> versioninfo()
Julia Version 1.9.0-DEV.1566
Commit ea991745a99 (2022-10-10 13:10 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 16 × AMD Ryzen 7 1700 Eight-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, znver1)
  Threads: 1 on 16 virtual cores
Environment:
  JULIA_EDITOR = code
  JULIA_NUM_THREADS = 1

pkg> st Aqua
Status `Project.toml`
  [4c88cf16] Aqua v0.5.5
```

### PR DESCRIPTION
**Pull request by jakobnissen**
Explain how seemingly well define typevars can result in unbound args in the presence of Vararg, and also display the offending methods in a more readable way.
Closes #87

### CODE REVIEW

**Comment by Roger-luo** [comment]
the PR looks good to me, but I'm not sure who has merge rights now, given @tkf seems not active on GitHub anymore...

**Comment by fingolfin** [comment]
@DilumAluthge made a release last week, so at the very least he seems to have commit rights. But I am not sure if he also is interested in maintaining the package overall (or if anyone is).

File: src/unbound_args.jl [hunk 1]
```diff
@@ -5,15 +5,34 @@ Test that all methods in `module` and its submodules do not have
...
+    @test isempty(unbounds)
 end
 
-function detect_unbound_args_recursively(m)
-    methods = []
-    walkmodules(m) do x
-        append!(methods, detect_unbound_args(x))
+# This was fixed some time between 1.4.2 and 1.6.7
+# https://github.com/JuliaLang/julia/pull/31972
+@static if VERSION < v"1.6.7"
```
Comment (fingolfin) [inline src/unbound_args.jl:28]: Just so I understand: this part of the patch is unrelated to the actual subject of the PR, right? It just disables an old workaround for newer Julia versions, yes?

Comment (fingolfin) [inline src/unbound_args.jl:28]: I just verified that it works in Julia 1.5.4 but crashes in 1.4.2
```suggestion
# This was fixed some time between 1.4.2 and 1.5.4
# https://github.com/JuliaLang/julia/pull/31972
@static if VERSION < v"1.5.4"
```

... not that it matters much either way ...

**Comment by fingolfin** [comment]
Thanks!

**Pull request merged**